Generals/Code/GameEngine/Source/GameLogic/Object/Object.cpp

```diff
@@ -2796,7 +2784,6 @@ void Object::onVeterancyLevelChanged( VeterancyLevel oldLevel, VeterancyLevel ne
 			clearWeaponBonusCondition(WEAPONBONUSCONDITION_VETERAN);
 			clearWeaponBonusCondition(WEAPONBONUSCONDITION_ELITE);
 			clearWeaponBonusCondition(WEAPONBONUSCONDITION_HERO);
-			doAnimation = FALSE;//... but not if somehow up to Regular
```

Comment:
Shouldn't this do `provideFeedback = FALSE` ?

Reply:
I suppose it could, but it's already covered by the `newLevel > oldLevel` part of the condition (and `oldLevel` is never expected to be less than `LEVEL_REGULAR`).

Reply:
Makes sense